engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1375,28 +1434,47 @@ private void processOperatorDeployStatus(final PTOperator oper, OperatorHeartbea
         break;
       case PENDING_UNDEPLOY:
         if (ds == null) {
+          long groupId = 0;
+          ContainerRedeployRequest request = getStopOperatorGroupId(oper, ohb);
+          if (request != null) {
+            groupId = request.getGroupId();
+            request.getOpeartorsTodeploy().add(oper.getId());
+          }
           // operator no longer deployed in container
-          recordEventAsync(new StramEvent.StopOperatorEvent(oper.getName(), oper.getId(), oper.getContainer().getExternalId()));
+          recordEventAsync(new StramEvent.StopOperatorEvent(oper.getName(), oper.getId(), oper.getContainer().getExternalId(), groupId));
           oper.setState(State.PENDING_DEPLOY);
-          sca.deployOpers.add(oper);
+          sca.deployOpers.put(oper, groupId);
         } else {
           // operator is currently deployed, request undeploy
           sca.undeployOpers.add(oper.getId());
           slowestUpstreamOp.remove(oper);
         }
         break;
       case PENDING_DEPLOY:
+        long eventGroupId = 0;
         if (ds == null) {
+          for (ContainerRedeployRequest requests : redeployRequests.values()) {
```

Comment:
This logic can be moved to the DeploymentGroupManager class I suggested elsewhere